src/components/PlayerCard/PlayerCard.js

```diff
@@ -0,0 +1,119 @@
+import cn from 'clsx';
+import { useMutation } from '@apollo/client';
+import { toast } from 'react-hot-toast';
+import { RiThumbUpLine } from 'react-icons/ri';
+import { IoMdThumbsUp } from 'react-icons/io';
+import { Text } from 'components';
+import { useAuthContext } from 'providers';
+import { UPDATE_LIKE } from 'constants/queries/queries';
+import { isVideoLiked } from 'utils/helperFuncs';
+import './PlayerCard.css';
+
+const PlayerCard = ({
+  video,
+  className,
+  refetchVideos,
+  cardActionProps,
+  children,
+  ellipsisText,
+}) => {
+  const { user } = useAuthContext();
+
+  const [updateLike] = useMutation(UPDATE_LIKE);
+
+  const liked = isVideoLiked(video?.likes, user?.id);
+
+  const handleLike = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (user?.id) {
+      const res = await updateLike({
+        variables: {
+          videoId: video?.id || '',
+        },
+      });
+      const { success } = res.data?.updateLike || false;
+      if (success) {
+        refetchVideos();
+      }
+      return null;
+    }
+
+    return toast.error('You must be logged in to like a video');
+  };
+
+  return (
+    <section className={cn('PlayerCard__root', className)}>
+      {children}
+      <div className="PlayerCard__contentWrapper">
+        <div className="d-flex content-between">
+          <div
+            role="button"
+            aria-hidden
```

Comment:
I wasn't aware of using the button role. On reading further, I found this -
"Adding `role="button"` tells assistive technology that the element is a button but provides no button functionality. Use `button` or `input` with `type="button"` instead."

Just curious, any specific reason to use `div` here?

Reply:
Thanks for pointing. Just to clear, it's allowed to  use div as button with role if any listener is attached to it. In my case, I have attached an onClick listener.